common/lib/client/paginatedresource.ts

```diff
@@ -0,0 +1,195 @@
+		this.rest = rest;
+		this.path = path;
+		this.headers = headers;
+		this.envelope = envelope ?? null;
+		this.bodyHandler = bodyHandler;
+		this.useHttpPaginatedResponse = useHttpPaginatedResponse || false;
+	}
+
+	get<T1, T2>(params: Record<string, T2>, callback: PaginatedResultCallback<T1>): void {
+		Resource.get(this.rest, this.path, this.headers, params, this.envelope, (err: ErrorInfo, body: unknown, headers: Record<string, string>, unpacked: boolean, statusCode: number) => {
+			this.handlePage(err, body, headers, unpacked, statusCode, callback);
+		});
+	}
+
+	delete<T1, T2>(params: Record<string, T2>, callback: PaginatedResultCallback<T1>): void {
+		Resource.delete(this.rest, this.path, this.headers, params, this.envelope, (err: ErrorInfo, body: unknown, headers: Record<string, string>, unpacked: boolean, statusCode: number) => {
+			this.handlePage(err, body, headers, unpacked, statusCode, callback);
+		});
+	}
+
+	post<T1, T2>(params: Record<string, T2>, body: unknown, callback: PaginatedResultCallback<T1>): void {
+		Resource.post(this.rest, this.path, body, this.headers, params, this.envelope, (err: ErrorInfo, resbody: unknown, headers: Record<string, string>, unpacked: boolean, statusCode: number) => {
+			if(callback) {
+				this.handlePage(err, resbody, headers, unpacked, statusCode, callback);
+			}
+		});
+	}
+
+	put<T1, T2>(params: Record<string, T2>, body: unknown, callback: PaginatedResultCallback<T1>): void {
+		Resource.put(this.rest, this.path, body, this.headers, params, this.envelope, (err: ErrorInfo, resbody: unknown, headers: Record<string, string>, unpacked: boolean, statusCode: number) => {
+			if(callback) {
+				this.handlePage(err, resbody, headers, unpacked, statusCode, callback);
+			}
+		});
+	}
+
+	patch<T1, T2>(params: Record<string, T2>, body: unknown, callback: PaginatedResultCallback<T1>): void {
+		Resource.patch(this.rest, this.path, body, this.headers, params, this.envelope, (err: ErrorInfo, resbody: unknown, headers: Record<string, string>, unpacked: boolean, statusCode: number) => {
+			if(callback) {
+				this.handlePage(err, resbody, headers, unpacked, statusCode, callback);
+			}
+		});
+	}
+
+	handlePage<T>(err: ErrorInfo, body: unknown, headers: Record<string, string>, unpacked: boolean, statusCode: number, callback: PaginatedResultCallback<T>): void {
+		if(err && returnErrOnly(err, body, this.useHttpPaginatedResponse)) {
+			Logger.logAction(Logger.LOG_ERROR, 'PaginatedResource.handlePage()', 'Unexpected error getting resource: err = ' + Utils.inspectError(err));
+			callback(err);
+			return;
+		}
+		let items, linkHeader, relParams;
+		try {
+			items = this.bodyHandler(body, headers, unpacked);
+		} catch(e) {
+			/* If we got an error, the failure to parse the body is almost certainly
+			 * due to that, so cb with that in preference to the parse error */
```

Comment:
"so cb with that in preference to the parse error" I think there is a typo here

Reply:
cb is short for callback but I agree that it's still not worded great so I improved it in cdfa6f7

Reply:
Resolving this conversation on @ikbalkaya's behalf.